bundles/org.openhab.core.io.rest.auth/src/main/java/org/openhab/core/io/rest/auth/internal/AuthFilter.java

```diff
@@ -111,15 +139,33 @@ private SecurityContext authenticateBearerToken(String token) throws Authenticat
         }
     }
 
-    private SecurityContext authenticateUsernamePassword(String username, String password)
+    private SecurityContext authenticateUsernamePassword(String username, String password, String authCacheKey)
             throws AuthenticationException {
+        final ByteBuffer cacheKey = getCacheKey(authCacheKey);
+        final UserSecurityContext cachedValue = AuthFilter.authCache.get(cacheKey);
+        if (cachedValue != null) {
+            return cachedValue;
+        }
+
         UsernamePasswordCredentials credentials = new UsernamePasswordCredentials(username, password);
         Authentication auth = userRegistry.authenticate(credentials);
         User user = userRegistry.get(auth.getUsername());
         if (user == null) {
             throw new AuthenticationException("User not found in registry");
         }
-        return new UserSecurityContext(user, auth, "Basic");
+
+        // limit cache to a certain size
+        if (AuthFilter.authCache.size() > 20) {
+            Object remName = null;
+            for (Object obj : AuthFilter.authCache.keySet()) {
```

Comment:
I don't understand the logic here. What's the use of a for loop that immediately breaks in the first iteration?

Reply:
It's limiting the cache size to 20 entries by removing an object from the cache

Reply:
You do not need a for loop for that.